src/components/provider/QueryModal.tsx

```diff
@@ -113,6 +121,25 @@ export default function QueryModal({
           )}
         </>
       )}
+      <div className="flex">
+        <Button
+          isDisabled={
+            selection === 'permit'
+              ? !permitData?.permitName ||
+                !permitData?.permitSignature ||
+                !permitData?.publicAddress ||
+                !permitData?.permitName.trim() ||
+                !permitData?.permitSignature.trim() ||
+                !permitData?.publicAddress.trim()
```

Comment:
can you extract this into its own function? `selection === 'permit' ? checkPermit() : checkViewingKey()`